logstash-core/spec/logstash/settings/modules_spec.rb

```diff
@@ -29,17 +30,17 @@
       it "should convert password Strings to Password" do
         source = [{"var.kibana.password" => secret}]
         setting = subject.set(source)
-        expect(setting).to be_a(Array)
-        expect(setting.__class__).to eq(LogStash::Util::ModulesSettingArray)
+        expect(setting).to be_a(java.util.ArrayList)
+        expect(setting.class).to eq(LogStash::Util::ModulesSettingArray)
         expect(setting.first.fetch("var.kibana.password")).to be_a(LogStash::Util::Password)
         expect(setting.first.fetch("var.kibana.password").value).to eq(secret)
       end
 
       it 'should not wrap values that are already passwords' do
         source = [{"var.kibana.password" => LogStash::Util::Password.new(secret)}]
         setting = subject.set(source)
-        expect(setting).to be_a(Array)
-        expect(setting.__class__).to eq(LogStash::Util::ModulesSettingArray)
+        expect(setting).to be_a(java.util.ArrayList)
+        expect(setting.class).to eq(LogStash::Util::ModulesSettingArray)
         expect(setting.first.fetch("var.kibana.password")).to be_a(LogStash::Util::Password)
```

Comment:
we are still going to rely on "magic" JRuby `java.util.List` additions here instead of being explicit.
e.g. `settings.first` is used through the code but is not explicitly provided by the Java impl ...

should consider adding `getFirst()` method or change those places to use `[0]` depending whether we can allow an indexing failure if `size == 0`

Reply:
why do not leverage the JRuby work to make a Java List seems like a Ruby Array?

Reply:
for folks not that familiar with JRuby these changes might be adding another barrier to understand:
- they need to understand that Java methods are mapped in Ruby (acceptable since it's a Java class)
- on top of that there are JRuby's Ruby-like additions to make `java.util.List` a bit similar to `Array`

am all :ok: with these, given existing code-base (JRuby-like) "barriers", 
just wanted to point out that some of the Java class' interface used through LS is relying on is not explicitly there.
this will likely be addressed later as parts where the class is used are migrated to Java.

Reply:
in this case I agree with you, adding `getFirst` and `getLast` to `ModulesSettingsArray`, thanks to chime in